driver-sync/src/test/functional/com/mongodb/client/unified/EventMatcher.java

```diff
@@ -79,4 +89,83 @@ public void assertEventsEquality(final String client, final BsonArray expectedEv
         }
         context.pop();
     }
+
+    public void assertConnectionPoolEventsEquality(final String client, final BsonArray expectedEventDocuments, final List<Object> events) {
+        context.push(ContextElement.ofConnectionPoolEvents(client, expectedEventDocuments, events));
+        assertEquals(context.getMessage("Number of events must be the same"), expectedEventDocuments.size(), events.size());
+
+        for (int i = 0; i < events.size(); i++) {
+            Object actual = events.get(i);
+            BsonDocument expectedEventDocument = expectedEventDocuments.get(i).asDocument();
+            String eventType = expectedEventDocument.getFirstKey();
+            context.push(ContextElement.ofConnectionPoolEvent(expectedEventDocument, actual, i));
+            BsonDocument expected = expectedEventDocument.getDocument(eventType);
+
+            if (actual.getClass().equals(ConnectionPoolCreatedEvent.class)) {
+                assertEquals(context.getMessage("Expected ConnectionPoolOpenedEvent"), eventType, "poolCreatedEvent");
```

Comment:
Right, but it's just for the context message, which requires it.  What would you like me to change it to?

Reply:
> What would you like me to change it to?

Something either empty or auto-generated so that it does not gets in the way of implementing https://github.com/mongodb/mongo-java-driver/pull/696/files#r618728979. I paid attention to these messages because of the aforementioned proposed simplification.

Reply:
Done